scripts/capture_and_validate.py

```diff
@@ -0,0 +1,144 @@
+import os
+import argparse
+import subprocess
+import glob as gl
+import re
+import hashlib
+import numpy as np
+
+# args:
+#   Enqueue number
+#   Link to CLI
+#   Link to app
+#   Arguments to app
+parser = argparse.ArgumentParser(description='Helper script to extract and validate single kernel from OpenCL app.')
+
+# Users can specify either an exact kernel name, xor an enqueue number
+group = parser.add_mutually_exclusive_group()
+group.add_argument('-kernel-name', '--name', dest='kernel_name',
+                    help='Name of the kernel that you want to capture, will capture the first time that sees this kernel', default=None)
+group.add_argument('-enqueue_number', '--num', type=int, dest='enqueue_number',
+                    help='The enqueue number that should be captured', default=-1)
+
+parser.add_argument('-cli', dest='cli_location',
+                    help='Location of the OpenCL-Intercept-Layer')
+parser.add_argument('-program', '--p', dest='app_location',
+                    help='Location of the app from which should be captured')
+parser.add_argument('-args', '--a', nargs=argparse.REMAINDER, default=[], dest='args')
+
+args = parser.parse_args()
+
+intercept_location_win = "C:\Intel\CLIntercept_Dump\\"
+intercept_location_posix = "~/CLIntercept_Dump/"
+intercept_location = ""
+delim = ""
+
+if os.name == 'nt':
+    intercept_location = intercept_location_win
+    delim = "\\"
+elif os.name == 'posix':
+    intercept_location = intercept_location_posix
+    delim = "/"
+else:
+    print("Unknown platform, exiting!")
+    exit()
+
+app_name = args.app_location[args.app_location.rfind(delim) + 1:]
+
+os.environ['CLI_InitializeBuffers'] = str(1)
+os.environ['CLI_AppendBuildOptions'] = "-cl-kernel-arg-info"
+
+if args.enqueue_number != -1:
+    os.environ['CLI_DumpReplayKernelEnqueue'] = str(args.enqueue_number)
+    os.environ['CLI_DumpBuffersMinEnqueue'] = str(args.enqueue_number)
+    os.environ['CLI_DumpBuffersMaxEnqueue'] = str(args.enqueue_number)
+    os.environ['CLI_DumpImagesMinEnqueue'] = str(args.enqueue_number)
+    os.environ['CLI_DumpImagesMaxEnqueue'] = str(args.enqueue_number)
+else:
+    os.environ['CLI_DumpReplayKernelName'] = args.kernel_name
+
+# Run ./cliloader, dumping via either enqueue number or the kernel name
+command = [args.cli_location, args.app_location]
+command.extend(args.args)
+
+subprocess.run(command)
+
+if args.enqueue_number != -1:
+    replay_location = os.path.join(intercept_location, app_name, "Replay", "Enqueue_" + str(args.enqueue_number), "")
+else:
+    replay_location = os.path.join(intercept_location, app_name, "Replay", "Enqueue_" + args.kernel_name, "")
+replay_location = os.path.expanduser(replay_location)
+
+# Run extracted kernel to dump output buffers
+os.chdir(replay_location)
+subprocess.run(["python", "run.py"])
```

Comment:
Minor: This didn't work on my system because my python executable is "python3".

Reply:
Now changed to use `sys.executable` to get the python executable, that should fix this :)